# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347946350 14400
#      Tue Sep 18 01:32:30 2012 -0400
# Node ID f52a62a6db3aeb2e42dd60a643cbbad24a1d2ce0
# Parent  7b69cd89868cbbc31b7d8cccf0e8757837470d26
eliminate copy and paste copyright info in GUI about info box

* libgui/src/main-window.cc (main_window::show_about_octave):
Use OCTAVE_STARTUP_MESSAGE instead of duplicating the text.
* libgui/src/module.mk (src_libgui_src_la_CPPFLAGS): Include
-I$(top_builddir)/libinterp in the list.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -38,19 +38,21 @@ along with Octave; see the file COPYING.
 #include <QIcon>
 
 #include "file-editor.h"
 #include "main-window.h"
 #include "octave-link.h"
 #include "settings-dialog.h"
 
 #include "debug.h"
+#include "defaults.h"
 #include "load-save.h"
 #include "toplev.h"
 #include "variables.h"
+#include "version.h"
 
 #include "cmd-hist.h"
 #include "oct-env.h"
 
 main_window::main_window (QWidget *p)
   : QMainWindow (p)
 {
   // We have to set up all our windows, before we finally launch octave.
@@ -371,32 +373,17 @@ void
 main_window::debug_quit ()
 {
   octave_link::post_event (this, &main_window::debug_quit_callback);
 }
 
 void
 main_window::show_about_octave ()
 {
-  QString message =
-    "GNU Octave\n"
-    "Copyright (C) 2009 John W. Eaton and others.\n"
-    "This is free software; see the source code for copying conditions."
-    "There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or"
-    "FITNESS FOR A PARTICULAR PURPOSE.  For details, type `warranty'.\n"
-    "\n"
-    "Additional information about Octave is available at http://www.octave.org.\n"
-    "\n"
-    "Please contribute if you find this software useful."
-    "For more information, visit http://www.octave.org/help-wanted.html\n"
-    "\n"
-    "Report bugs to <bug@octave.org> (but first, please read"
-    "http://www.octave.org/bugs.html to learn how to write a helpful report).\n"
-    "\n"
-    "For information about changes from previous versions, type `news'.\n";
+  QString message = OCTAVE_STARTUP_MESSAGE;
 
   QMessageBox::about (this, tr ("About Octave"), message);
 }
 
 void
 main_window::closeEvent (QCloseEvent *e)
 {
   e->ignore ();
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -136,17 +136,17 @@ src_libgui_src_la_CPPFLAGS = \
   -I$(srcdir)/src/octave-adapter \
   -I$(srcdir)/src/qtinfo \
   -I$(top_srcdir)/liboctave/cruft/misc \
   -I$(top_srcdir)/liboctave/array \
   -I$(top_builddir)/liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
   -I$(top_builddir)/liboctave/operators -I$(top_srcdir)/liboctave/operators \
   -I$(top_srcdir)/liboctave/system \
   -I$(top_srcdir)/liboctave/util \
-  -I$(top_srcdir)/libinterp \
+  -I$(top_builddir)/libinterp -I$(top_srcdir)/libinterp \
   -I$(top_builddir)/libinterp/parse-tree -I$(top_srcdir)/libinterp/parse-tree \
   -I$(top_builddir)/libinterp/interp-core -I$(top_srcdir)/libinterp/interp-core \
   -I$(top_builddir)/libinterp/interpfcn -I$(top_srcdir)/libinterp/interpfcn \
   -I$(top_srcdir)/libinterp/octave-value
 
 src_libgui_src_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 src_libgui_src_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
